### PR TITLE
drools-6847 refactoring ResourceToPkgDescrMapper

### PR DESCRIPTION
**Pull request by nprentza**
**Thank you for submitting this pull request**

**NOTE!:** Double check the target branch for this PR.
The default is `main` so it will target Drools 8 / Kogito / Optaplanner 8.
If this PR is not strictly related to drools and kogito project in `drools.git`, it should probably target `7.x`as a branch

**Ports** If a forward-port or a backport is needed, paste the forward port PR here

[link](https://www.example.com)

**JIRA**: _(please edit the JIRA link if it exists)_

[link](https://www.example.com)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>

* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`

* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-compiler/src/main/java/org/drools/compiler/builder/impl/ProcessorXml.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+public class ProcessorXml extends Processor {
+
+    public ProcessorXml(KnowledgeBuilderConfigurationImpl configuration){
+        super(configuration);
+    }
+
+    public PackageDescr process(Resource resource) throws DroolsParserException, IOException {
+        final XmlPackageReader xmlReader = new XmlPackageReader(this.configuration.getSemanticModules());
+        xmlReader.getParser().setClassLoader(this.configuration.getClassLoader()); //(this.rootClassLoader);
```
Comment (evacchi) [inline drools-compiler/src/main/java/org/drools/compiler/builder/impl/ProcessorXml.java]: it's ok if you remove this comment

File: drools-compiler/src/main/java/org/drools/compiler/builder/impl/Processor.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import org.drools.compiler.lang.dsl.DefaultExpander;
+import org.drools.core.util.IoUtils;
+import org.drools.drl.ast.descr.PackageDescr;
+import org.kie.api.builder.ReleaseId;
+import org.kie.api.io.Resource;
+import org.kie.internal.builder.KnowledgeBuilderResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
```
Comment (evacchi) [inline drools-compiler/src/main/java/org/drools/compiler/builder/impl/Processor.java]: star (*) imports should be avoided in our codebase, make sure you configure your environment properly

Screen Shot 2022-04-19 at 12 06 49
 
 
 I think you can load this config file in your IDE https://github.com/kiegroup/droolsjbpm-build-bootstrap/tree/main/ide-configuration/intellij-configuration/code-style

File: drools-compiler/src/main/java/org/drools/compiler/builder/impl/CompositeKnowledgeBuilderImpl.java [hunk 3]
```diff
@@ -332,10 +332,45 @@ private interface ResourceBuilder {
...
+        };
+        // Drools-6847
+        //ResourceToPkgDescrMapper DSLR_TO_PKG_DESCR = ( kBuilder, resourceDescr ) -> kBuilder.dslrToPackageDescr(resourceDescr.resource);
+        ResourceToPkgDescrMapper DSLR_TO_PKG_DESCR = ( kBuilder, resourceDescr ) ->{
+            ProcessorDslr processor = new ProcessorDslr(kBuilder.getBuilderConfiguration());
+            PackageDescr pkg = processor.process(resourceDescr.resource);
+            processor.getResults().forEach(kBuilder::addBuilderResult);
+            return pkg;
+        };
+        // Drools-6847
```
Comment (evacchi) [inline drools-compiler/src/main/java/org/drools/compiler/builder/impl/CompositeKnowledgeBuilderImpl.java]: you can probably remove these comments

File: drools-compiler/src/main/java/org/drools/compiler/builder/impl/Processor.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
+package org.drools.compiler.builder.impl;
```
Comment (evacchi) [inline drools-compiler/src/main/java/org/drools/compiler/builder/impl/Processor.java:16]: When a new class/file is created you should ensure a copyright header is present, like so https://github.com/kiegroup/drools/blob/main/drools-core/src/main/java/org/drools/core/BeliefSystemType.java#L1-L15

You can also make it automated e.g. see https://www.jetbrains.com/help/idea/copyright.html

**Review by evacchi — COMMENTED** [review]
Conflicting PR. Make sure you rebase or merge `main`. Otherwise it looks good.

**Comment by nprentza** [comment]
new PR #4321

**Pull request closed without merging**